.github/workflows/build-test-image.yml

```diff
@@ -0,0 +1,44 @@
+name: Build Test Image
+
+on:
+  workflow_run:
+    workflows: [Build/Test]
+    types: [completed]
+
+concurrency:
+  group: build-opencost-test
+  cancel-in-progress: false
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+
+jobs:
+  build-and-publish-opencost:
+    runs-on: ubuntu-latest
+    if: |
+      github.event.workflow_run.conclusion == 'success' && 
+      contains(github.event.workflow_run.head_branch, 'gh-readonly-queue')
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
```

Comment:
checkout the specific head sha, this will check out a merge commit by default which isn't what we do, we squash